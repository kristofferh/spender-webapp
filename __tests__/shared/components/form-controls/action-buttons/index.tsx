import React from "react";
import renderer from "react-test-renderer";

import ActionButtons from "shared/components/form-controls/action-buttons";

describe("Create ActionButtons snapshot", () => {
  it("should render and match no props snapshot", () => {
    const tree = renderer.create(<ActionButtons />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render and match non-default primary label snapshot", () => {
    const tree = renderer
      .create(<ActionButtons primaryActionLabel="Next" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render and match a secondary action snapshot", () => {
    const tree = renderer
      .create(
        <ActionButtons secondaryAction={true} secondaryActionLabel="Delete" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
