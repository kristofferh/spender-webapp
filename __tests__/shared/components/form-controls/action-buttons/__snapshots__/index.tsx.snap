// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create ActionButtons snapshot should render and match a secondary action snapshot 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 40px;
  width: 100%;
  border: 0;
  background-color: #101010;
  color: #ffffff;
  padding: 10px 20px;
  border-radius: 4px;
  -webkit-transition: background-color 0.2s ease,box-shadow 0.2s ease,color 0.3s ease;
  transition: background-color 0.2s ease,box-shadow 0.2s ease,color 0.3s ease;
  cursor: pointer;
  font-family: inherit;
  font-size: inherit;
  pointer-events: all;
  margin-left: auto;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  padding: 0 50px;
}

.emotion-5:active,
.emotion-5:hover {
  background-color: #000000;
  color: #ffffff;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0 0 0 3px #66afe965;
}

.emotion-2 {
  width: 20px;
  height: 20px;
  margin-right: -30px;
  margin-left: 10px;
}

.emotion-0 {
  font-size: 15px;
  cursor: pointer;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: bold;
}

<footer
  className="emotion-9 emotion-10"
>
  <div
    className="emotion-7 emotion-8"
  >
    <a
      className="emotion-0 emotion-1"
    >
      Delete
    </a>
    <button
      className="emotion-4 emotion-5 emotion-6"
    >
      Submit
      <div
        className="emotion-2 emotion-3"
      />
    </button>
  </div>
</footer>
`;

exports[`Create ActionButtons snapshot should render and match no props snapshot 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 40px;
  width: 100%;
  border: 0;
  background-color: #101010;
  color: #ffffff;
  padding: 10px 20px;
  border-radius: 4px;
  -webkit-transition: background-color 0.2s ease,box-shadow 0.2s ease,color 0.3s ease;
  transition: background-color 0.2s ease,box-shadow 0.2s ease,color 0.3s ease;
  cursor: pointer;
  font-family: inherit;
  font-size: inherit;
  pointer-events: all;
  margin-left: auto;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  padding: 0 50px;
}

.emotion-3:active,
.emotion-3:hover {
  background-color: #000000;
  color: #ffffff;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 0 3px #66afe965;
}

.emotion-0 {
  width: 20px;
  height: 20px;
  margin-right: -30px;
  margin-left: 10px;
}

<footer
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-5 emotion-6"
  >
    <button
      className="emotion-2 emotion-3 emotion-4"
    >
      Submit
      <div
        className="emotion-0 emotion-1"
      />
    </button>
  </div>
</footer>
`;

exports[`Create ActionButtons snapshot should render and match non-default primary label snapshot 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 40px;
  width: 100%;
  border: 0;
  background-color: #101010;
  color: #ffffff;
  padding: 10px 20px;
  border-radius: 4px;
  -webkit-transition: background-color 0.2s ease,box-shadow 0.2s ease,color 0.3s ease;
  transition: background-color 0.2s ease,box-shadow 0.2s ease,color 0.3s ease;
  cursor: pointer;
  font-family: inherit;
  font-size: inherit;
  pointer-events: all;
  margin-left: auto;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  padding: 0 50px;
}

.emotion-3:active,
.emotion-3:hover {
  background-color: #000000;
  color: #ffffff;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 0 3px #66afe965;
}

.emotion-0 {
  width: 20px;
  height: 20px;
  margin-right: -30px;
  margin-left: 10px;
}

<footer
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-5 emotion-6"
  >
    <button
      className="emotion-2 emotion-3 emotion-4"
    >
      Next
      <div
        className="emotion-0 emotion-1"
      />
    </button>
  </div>
</footer>
`;
